test/functional/npcontainer.spec.ts

```diff
@@ -0,0 +1,98 @@
+import assert from "assert";
+import { CosmosClient, Constants, Container } from "../../dist-esm";
+import { removeAllDatabases, getTestContainer } from "../common/TestHelpers";
+import { endpoint, masterKey } from "../common/_testConfig";
+import { ResourceType, HTTPMethod } from "../../dist-esm/common";
+
+const legacyClient = new CosmosClient({
+  endpoint,
+  key: masterKey,
+  plugins: [
+    {
+      on: "request",
+      plugin: (context, next) => {
+        // Intercepts the API request to create a non-partitioned container using an old API version
+        if (context.resourceType === ResourceType.container && context.method === HTTPMethod.post) {
+          context.body = JSON.stringify({ id: JSON.parse(context.body).id });
```

Comment:
what is this doing? Dropping everything except the name of the container? To avoid weird differences in the API versions?

Reply:
Yes. We now default to `_partitionKey` inside the SDK, so need to "undo" that for this call. IMO it would be nice if the backend accepted just an `id` and defaulted `_partitionKey` there